pkg/sql/schemachanger/scexec/scmutationexec/references.go

```diff
@@ -89,6 +90,80 @@ func (m *visitor) RemoveCheckConstraint(ctx context.Context, op scop.RemoveCheck
 	return nil
 }
 
+func (m *visitor) AddCheckConstraint(ctx context.Context, op scop.AddCheckConstraint) error {
+	tbl, err := m.checkOutTable(ctx, op.TableID)
+	if err != nil || tbl.Dropped() {
+		return err
+	}
+	if op.ConstraintID >= tbl.NextConstraintID {
+		tbl.NextConstraintID = op.ConstraintID + 1
+	}
+
+	// We should have already validated that the check constraint
+	// is syntactically valid in the builder, so we just need to
+	// enqueue it to the descriptor's mutation slice.
+	tbl.AddCheckMutation(&descpb.TableDescriptor_CheckConstraint{
+		Expr:         string(op.Expr),
+		Name:         tabledesc.ConstraintNamePlaceholder(op.ConstraintID),
+		Validity:     descpb.ConstraintValidity_Validating,
+		ColumnIDs:    op.ColumnIDs,
+		Hidden:       true,
+		ConstraintID: op.ConstraintID,
+	}, descpb.DescriptorMutation_ADD)
+	return nil
+}
+
+func (m *visitor) MakeAddedCheckConstraintPublic(
+	ctx context.Context, op scop.MakeAddedCheckConstraintPublic,
+) error {
+	tbl, err := m.checkOutTable(ctx, op.TableID)
+	if err != nil || tbl.Dropped() {
+		return err
+	}
+
+	var found bool
+	for idx, mutation := range tbl.Mutations {
+		if c := mutation.GetConstraint(); c != nil &&
+			c.ConstraintType == descpb.ConstraintToUpdate_CHECK &&
+			c.Check.ConstraintID == op.ConstraintID {
+			// Add the check to the public `Checks` slice
+			tbl.Checks = append(tbl.Checks, &c.Check)
+
+			// Remove the mutation from the mutation slice. The `MakeMutationComplete`
+			// call will also mark the above added check as VALIDATED.
+			// If this is a rollback of a drop, we are trying to add the index back,
+			// so swap the direction before making it complete.
+			mutation.Direction = descpb.DescriptorMutation_ADD
+			err = tbl.MakeMutationComplete(mutation)
+			if err != nil {
+				return err
+			}
+			tbl.Mutations = append(tbl.Mutations[:idx], tbl.Mutations[idx+1:]...)
+
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		return errors.AssertionFailedf("failed to find check constraint %d in table %q (%d)",
+			op.ConstraintID, tbl.GetName(), tbl.GetID())
+	}
+
+	if len(tbl.Mutations) == 0 {
+		tbl.Mutations = nil
+	}
+
+	return nil
+}
+
+func (m *visitor) MakeDroppedCheckConstraintValidated(
+	ctx context.Context, constraint scop.MakeDroppedCheckConstraintValidated,
+) error {
+	// TODO (xiang): implement this
+	return nil
+}
```

Comment:
Does this op even need to exist? If it's not going anything, I'd rather it didn't.